src/backend/distributed/executor/local_executor.c

```diff
@@ -175,6 +178,120 @@ ExecuteLocalTaskList(CitusScanState *scanState, List *taskList)
 }
 
 
+/*
+ * LocalTaskPlannedStmt creates the PlannedStmt for the input task.
+ *
+ * The function behaves differently when the task->queryString is avaliable
+ * or not. See the comments in the function for the details.
+ */
+static Query *
+LocalShardQuery(Query *workerJobQuery, Task *task, ParamListInfo boundParams,
+				int *numParams, Oid **parameterTypes)
+{
+	/*
+	 * For performance reasons, the queryString is not generated for
+	 * local-fast path queries. Instead, we simply update the shard
+	 * references to the actual shard relations.
+	 *
+	 * Otherwise, like multi-shard queries going through the local execution,
+	 * we'd need to parse the queryString back to Query before passing it
+	 * to the planner,
+	 */
+	if (task->query != NULL)
+	{
+		Query *shardQuery = copyObject(workerJobQuery);
```

Comment:
Why do we copy from workerJobQuery, and not use `task->query`?

Reply:
Changed this now since it's more correct.